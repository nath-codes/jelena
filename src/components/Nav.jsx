import React from "react";
import styled from "styled-components";
import Link from "./Link";
import spacer from "../mixins/spacer";

const StyledNavItem = Link.extend`
  font-family: ${props => props.theme.fonts.heading};
  font-size: ${props => spacer(1, props.theme.breakpoint)};
  margin-right: ${props => spacer(1, props.theme.breakpoint)};

  &:last-child {
    margin-right: 0;
  }
`;

const StyledNav = styled.nav``;

const Nav = () => (
  <StyledNav>
    <StyledNavItem href="#rsvp">RSVP</StyledNavItem>
    <StyledNavItem href="#location">Location</StyledNavItem>
    <StyledNavItem href="#timetable">Timetable</StyledNavItem>
    <StyledNavItem href="#gifts">Gifts</StyledNavItem>
  </StyledNav>
);

export default Nav;
