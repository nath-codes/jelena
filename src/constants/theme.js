export default {
  fonts: {
    heading: "bebas-neue",
    body: "montserrat"
  },
  colors: {
    delftBlue: "#1b56c5",
  }
};
