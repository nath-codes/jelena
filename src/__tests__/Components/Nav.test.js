import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";
import Nav from "../../components/Nav";
import theme from "../../constants/theme";

describe("<Nav />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Nav />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
