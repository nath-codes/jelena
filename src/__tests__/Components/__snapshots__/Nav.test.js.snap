// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders correctly 1`] = `
.c0 {
  color: #1b56c5;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: bebas-neue;
  font-size: 1.2rem;
  margin-right: 1.2rem;
}

.c0::before {
  background-color: #fff;
  bottom: -3px;
  content: "";
  height: 2px;
  left: 0;
  position: absolute;
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.4,0,0.2,1);
  -webkit-transition: transform 0.3s cubic-bezier(0.4,0,0.2,1);
  transition: transform 0.3s cubic-bezier(0.4,0,0.2,1);
  width: 100%;
}

.c0:hover::before {
  background-color: #1b56c5;
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.c0:last-child {
  margin-right: 0;
}

<nav
  className=""
>
  <a
    className="c0"
    href="#rsvp"
  >
    RSVP
  </a>
  <a
    className="c0"
    href="#location"
  >
    Location
  </a>
  <a
    className="c0"
    href="#timetable"
  >
    Timetable
  </a>
  <a
    className="c0"
    href="#gifts"
  >
    Gifts
  </a>
</nav>
`;
